crates/edr_rpc_server/src/node.rs

```diff
@@ -437,4 +618,548 @@ mod tests {
+
+        let block_and_total_difficulty = fixture
+            .node
+            .block_and_total_difficulty_by_block_spec(&BlockSpec::Tag(BlockTag::Earliest))
+            .await
+            .unwrap();
+
+        let block_hash = block_and_total_difficulty.block.header().hash();
+
+        assert_block_hash(&fixture, &block_hash, None).await;
+        assert_block_hash(&fixture, &block_hash, Some(true)).await;
+        assert_block_hash(&fixture, &block_hash, Some(false)).await;
+
+        let non_existing_block = fixture
+            .node
+            .block_and_total_difficulty_by_block_spec(&BlockSpec::Eip1898(Eip1898BlockSpec::Hash {
+                block_hash: B256::zero(),
+                require_canonical: None,
+            }))
+            .await;
+
+        assert_error!(
+            non_existing_block,
+            NodeError::UnknownBlockHash { block_hash } => assert_eq!(block_hash, B256::zero())
+        );
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn block_and_total_difficulty_by_block_spec_total_difficulty() -> Result<()> {
+        let fixture = NodeTestFixture::new().await?;
+
+        fixture.node.lock_data().await.mine_block(None).await?;
+        fixture.node.lock_data().await.mine_block(None).await?;
+
+        let block_and_total_difficulty0 = fixture
+            .node
+            .block_and_total_difficulty_by_block_spec(&BlockSpec::Number(0))
+            .await
+            .unwrap();
+
+        let block_and_total_difficulty1 = fixture
+            .node
+            .block_and_total_difficulty_by_block_spec(&BlockSpec::Number(1))
+            .await
+            .unwrap();
+
+        assert_eq!(
+            block_and_total_difficulty1.total_difficulty,
+            block_and_total_difficulty0.total_difficulty
+                + block_and_total_difficulty1.block.header().difficulty
+        );
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn block_and_total_difficulty_by_hash() -> Result<()> {
+        let fixture = NodeTestFixture::new().await?;
+
+        let block_and_total_difficulty = fixture
+            .node
+            .block_and_total_difficulty_by_block_spec(&BlockSpec::Tag(BlockTag::Earliest))
+            .await
+            .unwrap();
+
+        let block_hash = block_and_total_difficulty.block.header().hash();
+
+        let block_and_total_difficulty = fixture
+            .node
+            .block_and_total_difficulty_by_hash(&block_hash)
+            .await
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(block_and_total_difficulty.block.header().hash(), block_hash);
+
+        let non_existing_block = fixture
+            .node
+            .block_and_total_difficulty_by_hash(&B256::zero())
+            .await
+            .unwrap();
+
+        assert!(non_existing_block.is_none());
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn block_and_total_difficulty_by_hash_total_difficulty() -> Result<()> {
```

Comment:
This feels excessive as the unit tests for the blockchain implementations already test this.

Is there a more minimal way of testing the code for this function?